File: 907-koko-eating-bananas/koko-eating-bananas.cs
public class Solution {
    public int MinEatingSpeed(int[] piles, int h) {
        int n = piles.Length;

        int l = 1, r = piles.Max();
        int res = r;
        while(l <= r)
        {
            int mid = (r + l) / 2;
            long sum = 0;
            foreach (int p in piles) {
                sum += (int)Math.Ceiling((double)p / mid);
            }
            if (sum <= h) {
                res = mid;
                r = mid - 1;
            } else {
                l = mid + 1;
            }
        }

        return res;
    }
}

File: 907-koko-eating-bananas/koko-eating-bananas.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int n = piles.size();

        int left = 1;int right = *max_element(piles.begin(), piles.end());
        int minTime = INT_MAX;

        while(left <= right)
        {
            int mid = left + (right - left) / 2;
            double curTime = 0;
            for(int pile : piles)
            {
                curTime += ceil((double)pile / mid);
            }
            if(curTime <= h)
            {
                minTime = min(mid, minTime);
            }

            if(curTime <= h)
            {
                right = mid - 1;
            }
            else
            {
                left = mid + 1;
            }
        }

        return minTime;
    }
};